include/enumerations/material_definitions.hpp

```diff
@@ -0,0 +1,166 @@
+#pragma once
```

Comment:
I think this file needs more explanation. 
A small paragraph at the top explaining the goal of this long list of macros.
Looking at this file it is not clear which macro is important and which one is not.
You explained this to us, and we understand what's important but if I had to look 
at this in two months it would take me a day to realize what's happening.

I think this also deserves a small section in the developers docs.